commit b00916b189d13a615ff05c9242201135992fcda3
Author: Kees Cook <kees.cook@canonical.com>
Date:   Mon Oct 11 12:23:25 2010 -0700

    net: clear heap allocations for privileged ethtool actions
    
    Several other ethtool functions leave heap uncleared (potentially) by
    drivers. Some interfaces appear safe (eeprom, etc), in that the sizes
    are well controlled. In some situations (e.g. unchecked error conditions),
    the heap will remain unchanged in areas before copying back to userspace.
    Note that these are less of an issue since these all require CAP_NET_ADMIN.
    
    Cc: stable@kernel.org
    Signed-off-by: Kees Cook <kees.cook@canonical.com>
    Acked-by: Ben Hutchings <bhutchings@solarflare.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    [dannf: backported to Debian's 2.6.26]

diff -urpN linux-source-2.6.26.orig/net/core/ethtool.c linux-source-2.6.26/net/core/ethtool.c
--- linux-source-2.6.26.orig/net/core/ethtool.c	2008-07-13 15:51:29.000000000 -0600
+++ linux-source-2.6.26/net/core/ethtool.c	2011-03-30 22:30:58.796187153 -0600
@@ -226,7 +226,7 @@ static int ethtool_get_regs(struct net_d
 	if (regs.len > reglen)
 		regs.len = reglen;
 
-	regbuf = kmalloc(reglen, GFP_USER);
+	regbuf = kzalloc(reglen, GFP_USER);
 	if (!regbuf)
 		return -ENOMEM;
 
